### ISSUE TITLE
`conda env create` ignoring `-n` flag

### ISSUE THREAD

**Issue by bow** [conda/conda#2995]
Hi Conda developers,

I stumbled on an issue recently where I stumbled on this error when running `conda env create`:

```
Error: prefix already exists: /usr/local/sasc/miniconda3
```

This seems like I'm trying to create an env with an existing name. But the path shown there is my root conda installation, not any package name.

Here's how I got there: I am inside a directory with nothing but an `environment.yml` file. If the `name` attribute inside the YAML file is defined, `conda env create` works great. However, if `name` is not defined, I got the error. What's more, if I tried supplying the name via the `-n` flag, I still get the error.

Is this intended behavior? If so, may I request that the error message be at least changed (something like, `required 'name' attribute is not found in the YAML file`). I wasn't sure what the error was about since the current error message show the root path.

If this is not intended behavior, I would imagine using the `-n` flag should work when `name:` is not present in the YAML file. If both are present and the values are different, there should then be a warning or error message saying that there are conflicting environment name values.

Here's my `conda info` output for reference:

```
Current conda install:

             platform : linux-64
        conda version : 4.1.5
    conda-env version : 2.5.0
  conda-build version : not installed
       python version : 3.5.1.final.0
     requests version : 2.10.0
     root environment : /usr/local/sasc/miniconda3  (writable)
  default environment : /usr/local/sasc/miniconda3
     envs directories : /usr/local/sasc/miniconda3/envs
        package cache : /usr/local/sasc/miniconda3/pkgs
         channel URLs : https://conda.anaconda.org/bioconda/linux-64/
                        https://conda.anaconda.org/bioconda/noarch/
                        https://conda.anaconda.org/r/linux-64/
                        https://conda.anaconda.org/r/noarch/
                        https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /home/warindrarto/.condarc
         offline mode : False
    is foreign system : False
```

Thanks in advance for looking into this :).

**Comment by dharhas** [comment]
This changed recently and was a rude shock since all my install scripts stopped working. It looks like the behaviour has been changed in recent versions and -n/--name is ignored completely. Previously -n/--name would override the name defined inside the yml file.

@ the developers, is there a reason for dropping the -n/--name option? 

This is the same issue raised here https://github.com/conda/conda-env/issues/257

another question, 

what is the reason for having two ways to create envs i.e. what is the difference between 

conda create -n myenv --file myfile.yml

and 

conda env create -n myenv --file myfile.yml

The myfile.yml syntax does not match between these two. Is there a reason for the two techniques or is it historical?

**Comment by jseabold** [comment]
Issue on the conda-env tracker https://github.com/conda/conda-env/issues/254

**Comment by jseabold** [comment]
Should be closable.

**Comment by kalefranz** [comment]
fixed with conda-env v2.5.2 available sometime today.
